### PR TITLE
Adds nextflow refined db

### PR DESCRIPTION
**Pull request by cgpu**
This PR adds the generation of the refined database as a Nextflow script.

### Overview

This PR updates the Nextflow specific files for the ORF calling modules.

### Description

The changes implement the following:

- [x] remove some boilerplate code to make the files cleaner, minimal
- [x] renames the custom config name to `nextflow.config` for simplicity
- [x] adds boolean param to enable execution of TransDecoder based on this flag (can be removed)
- [x]  modifies channel name to follow the convention **`ch_`<name of channel>** for readability eg. `ch_fasta`
- [x] adds temporary next process to display example of channel  i/o utilisation
- [x] updates the processes containers to  `gsheynkmanlab/transdecoder:5.5.0`
- [x] encourages use of `snake_case` in variable, process, channel names 

### How can I test this works?

Assuming you have cloned the `Long-Read-Proteogenomics` repo, run the following commands to checkout in the branch `cgpu-updates-orf-calling` that implements the change:

```bash

# Navigate into the repo folder
cd Long-Read-Proteogenomics

# Navigate to the branch that has the code
git checkout cgpu-updates-orf-calling

# Navigate to the folder with the Nextflow-ified standalone module:
cd modules/LR_ORFCalling
```

#### 1. **Positive** Nextflow test (should be completed successfully)

```groovy
 nextflow run lr_orfcalling.nf --fasta https://zenodo.org/record/4278034/files/toy_for_christina.fasta.txt
```

#### 2.  **Negative** Nextflow test (should be fail early)

Neglect providing the required parameter `--fasta`. This should fail early with an informative error message.

```groovy
 nextflow run lr_orfcalling.nf
```

### CODE REVIEW

File: modules/PG_RefinedDatabaseGeneration/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# nf-core style template for Dockerfile
+FROM continuumio/miniconda3:4.8.2
+LABEL description="Basic docker image containing util software requirements for the sheynkman-lab/Long-Read-Proteogenomics pipeline Refined Database Module"
+
```
Comment (cgpu) [inline modules/PG_RefinedDatabaseGeneration/Dockerfile:4]:
```Suggestion
ARG ENV_NAME="refined-database-generation"
```

File: modules/PG_RefinedDatabaseGeneration/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+# deep clean the apt cache to reduce image/layer size
+# ! Leave this as is for any module
+RUN apt-get update \
+ && apt-get install -y procps \
+ && apt-get clean -y && rm -rf /var/lib/apt/lists/*
+
+# ! Leave this as is for any module
+# Install the conda environment
+COPY environment.yml /
+RUN conda env create --quiet -f /environment.yml && conda clean -a
```
Comment (cgpu) [inline modules/PG_RefinedDatabaseGeneration/Dockerfile:15]:
```Suggestion
RUN conda env create --name ${ENV_NAME} --quiet --file /environment.yml && conda clean -a
```

File: modules/PG_RefinedDatabaseGeneration/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+# ! Update the envs/<name> based on the environment.yml file
+# Add conda installation dir to PATH (instead of doing 'conda activate')
+ENV PATH /opt/conda/envs/refined-database-generation/bin:$PATH
```
Comment (cgpu) [inline modules/PG_RefinedDatabaseGeneration/Dockerfile:19]:
```Suggestion
ENV PATH /opt/conda/envs/${ENV_NAME}/bin:$PATH
```

File: modules/PG_RefinedDatabaseGeneration/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+# Dump the details of the installed packages to a file for posterity
+RUN conda env export --name refined-database-generation > refined-database-generation.yml
```
Comment (cgpu) [inline modules/PG_RefinedDatabaseGeneration/Dockerfile:22]:
```Suggestion
RUN conda env export --name ${ENV_NAME} > ${ENV_NAME}.yml
```

File: modules/PG_RefinedDatabaseGeneration/refined_db_generation.nf [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    
+    Input files:
+      --orfs          path to ORF coordinate input file location
+      --seq           path to PacBio fasta sequence input file location
+      --redundant     name of output redundant accession file location   
+      --ctsv          name of output combined tsv file location
+      --cfasta        ...
+      --aggtsv 
+      --aggfasta 
+      --sample        name of unique identifier for the biosample
```
Comment (cgpu) [inline modules/PG_RefinedDatabaseGeneration/refined_db_generation.nf:31]: Update the help message parameters description refined_db_generation.nf

**Review by bj8th — APPROVED** [review]
reviewed and OK

**Pull request merged**